src/lib/storage/lz4/lz4_encoder.hpp

```diff
@@ -180,12 +196,14 @@ class LZ4Encoder : public SegmentEncoder<LZ4Encoder> {
     if (num_chars == 0) {
       auto empty_blocks = pmr_vector<pmr_vector<char>>{alloc};
       auto empty_dictionary = pmr_vector<char>{};
-      auto empty_offsets = pmr_vector<size_t>{};
       return std::allocate_shared<LZ4Segment<pmr_string>>(alloc, std::move(empty_blocks), std::move(null_values),
-                                                          std::move(empty_dictionary), std::move(empty_offsets),
-                                                          _block_size, 0u, 0u);
+                                                          std::move(empty_dictionary), nullptr, _block_size, 0u, 0u);
     }
 
+    // Compress the offsets with SIMDBP128 to reduce the memory footprint of the LZ4 segment.
+    auto vector_compressor = SimdBp128Compressor{};
```

Comment:
With the dictionary encoding, we allow different compressions to be used. If we bring in vector compression here, we should follow the pattern that it is decoupled from the chosen segment encoding.